lib/AST/Decl.cpp

```diff
@@ -4832,9 +4832,17 @@ static bool checkAccess(const DeclContext *useDC, const ValueDecl *VD,
     return srcPkg && usePkg && usePkg->isSamePackageAs(srcPkg);
   }
   case AccessLevel::Public:
-  case AccessLevel::Open:
+  case AccessLevel::Open: {
+    if (VD->isOperator() && VD->isSPI()) {
```

Comment:
It might be worth a comment explaining why operators need to be special cased here. I'm having trouble following the surrounding logic and why the existing centralized logic does not apply to them.